src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,56 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DATA_CAPACITY = 10;
+    private final Pair<K, V>[] data;
+    private int elementsCount;
+
+    public StorageImpl() {
+        this.data = new Pair[DATA_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = contains(key);
+        if (index > -1) {
```

Comment:
better eq comparing in such cases
```suggestion
        if (index != -1) {
```